core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -109,9 +150,33 @@
     this.regionalizedInstanceId = String.format("%s~%s", this.regionId, this.instanceId);
 
     this.apiClient = apiClient;
+    this.enableIamAuth = enableIamAuth;
     this.executor = executor;
     this.keyPair = keyPair;
 
+    if (enableIamAuth) {
+      // Get token source
+      String userTokenSourceFactoryClassName = System.getProperty(
+          TokenSourceFactory.TOKEN_SOURCE_FACTORY_PROPERTY);
+      TokenSourceFactory tokenSourceFactory;
+
+      if (userTokenSourceFactoryClassName != null) {
+        try {
+          tokenSourceFactory =
+              (TokenSourceFactory)
+                  Class.forName(userTokenSourceFactoryClassName).newInstance();
+        } catch (Exception err) {
+          throw new RuntimeException(err);
```

Comment:
Is `newInstance()` throwing any checked exceptions? If not, do we need to catch and rewrap here?

Reply:
`newInstance()` throws `InstantiationException` and `IllegalAccessException` and `forName()` throws  `ClassNotFoundException`